Automattic/wp-calypso
# NPS Survey: Whitelist sections where survey will be shown

This PR adds a whitelist to limit what sections the NPS survey notice will be shown on. The allowed sections are:

- 'discover'
- 'menus'
- 'people'
- 'plugins'
- 'posts-pages'
- 'reader'
- 'reader-activities'
- 'reader-list'
- 'reader-recommendations'
- 'reader-tags'
- 'settings'
- 'sharing'
- 'stats'

NPS Survey is only enabled in `development`, `wpcalypso`, and `staging`.

To test:

```
npm run test-client -- --grep "state nps-survey"
```

Use the trigger in `development` or `wpcalypso` to force the session to be eligible (normally, only 1 out of every 5000 sessions is). In the browser console, type:

```
npsSurvey();
```

If on a whitelisted section, the notice should be shown after a brief delay. If not on a whitelisted section, the notice should be shown after a brief delay once you navigate to a whitelisted section.

Once the notice has been shown once in the session (full page load), it will not be shown again during the same session.
